Added heading-2 through heading-6 mixins.

tiffanybbrown-va opened the pull request:
Closes department-of-veterans-affairs/vets.gov-team#760

annekainicUSDS committed:
First pass

annekainicUSDS committed:
Fix lint errors

annekainicUSDS committed:
Remove comments

annekainicUSDS committed:
Fix lint errors

annekainicUSDS committed:
remove commented out code

Commit:
Added heading-2 through headin-6 mixins.

Closes department-of-veterans-affairs/vets.gov-team#760

Commit:
Added heading-2 through headin-6 mixins.

Closes department-of-veterans-affairs/vets.gov-team#760

Commit:
Added heading-2 through headin-6 mixins.

Closes department-of-veterans-affairs/vets.gov-team#760

jcmeloni-usds commented:
LGTM

omgitsbillryan commented on src/sass/base/_va.scss:
```diff
@@ -289,45 +289,38 @@ hr {
-
-h1:first-child {
-  margin-top: 0;
+  @include heading-2;
```
Shouldn't this be `heading-1`?

omgitsbillryan commented on src/sass/base/_b-variables.scss:
```diff
@@ -7,7 +9,42 @@
+@mixin heading-6 {
+  font-size: 0.9em;
+  line-height: 1.5;
+  font-weight: bold;
```
Can `font-weight: bold;` be shared to avoid repetition?

omgitsbillryan commented on src/sass/base/_b-variables.scss:
```diff
@@ -7,7 +9,42 @@
 
+// $heading-1:
+@mixin heading-2 {
+  font-size: 1.2em;
```
Excuse my ignorance - shouldn't `heading-2` have a larger font size than 3?

jcmeloni-usds commented:
@tiffanybbrown-va and @melwoodard are there shenanigans here re: presentation vs semantics or is @omgitsbillryan on to something?

melwoodard commented:
I'd need to look closer at specific places we're using H2s, but there is some weirdness between H1/H2/H3 due to the fact that H2s have been used really sporadically. All of our content pages go right from H1 to H3, skipping H2 (not exactly sure why), so H2 probably only shows up in the apps and is being treated inconsistently.

jcmeloni-usds commented:
Sounds like shenanigans to me :)

tiffanybbrown-va commented:
Some additional context:

`h1` was originally reserved for the logo. At some point, this changed, and we started using `h1` throughout the content. Our `h1` and `h2` elements are styled the same. That's why there's a `heading-2` mixin, but no `heading-1` mixin. 

The mixin names and styles are pulled directly from whatever element name they were originally attached to. This is why the font size for `heading-3` is larger than that for `heading-2`. Made sense to me at the time. It apparently does not make sense to anyone else.

jcmeloni-usds commented:
I think it makes sense to me enough that I recognized it immediately as probably shenanigans. :)  If this all _works_ I'm apt to merge it in with a comment somewhere that mixin names and styles are pulled directly from whatever element name they were originally attached to, and someone should maybe take a look at that bigger issue someday...

melwoodard commented:
Agreed - we should put a pin in the fact that we're skipping a heading level (weird experience for screenreaders) and revisit it separately to get our headings right across the board.

Commit:
Moved mixins to _b-mixins.scss, updated style.scss to import it.

- Moved `media-maxwidth` from utils to mixins.
- Added note about heading size shenanigans.

Commit:
Moved mixins to _b-mixins.scss, updated style.scss to import it.

- Moved `media-maxwidth` from utils to mixins.
- Added note about heading size shenanigans.

Commit:
Merge branch 'department-of-veterans-affairs/vets.gov-team-760' of github.com-tiffanybbrown-va:department-of-veterans-affairs/vets-website into department-of-veterans-affairs/vets.gov-team-760

tiffanybbrown-va commented:
Not exactly sure why this test is failing, since I don't think it has anything to do with my changes to this PR 🤔

Commit:
Merge branch 'master' into department-of-veterans-affairs/vets.gov-team-760

cvalarida committed:
5490 previous benefits sponsor file number (#5291)

* Allow for file number in previous benefits sponsor info

* Hide the nested expanded border

* One more field to check for...

* Test ssn vs fn

annekainicUSDS committed:
Merge pull request #5129 from department-of-veterans-affairs/move-hca-to-new-endpoint-for-prod

Move hca to new endpoint for prod

Commit:
Merge branch 'department-of-veterans-affairs/vets.gov-team-760' of github.com-tiffanybbrown-va:department-of-veterans-affairs/vets-website into department-of-veterans-affairs/vets.gov-team-760

jcmeloni-usds commented:
All green now! Let's merge it.
